File: backend/src/main/java/com/clinicmanagement/consultation/ConsultationService.java
package com.clinicmanagement.consultation;

import com.clinicmanagement.common.dto.PageResponse;
import com.clinicmanagement.consultation.dto.ChangeConsultationStatusRequest;
import com.clinicmanagement.consultation.dto.ConsultationResponse;
import com.clinicmanagement.consultation.dto.CreateConsultationRequest;
import org.springframework.data.domain.Pageable;

public interface ConsultationService {

    PageResponse<ConsultationResponse> getAll(Long patientId, Long doctorId, String status, Pageable pageable);

    ConsultationResponse getById(Long id);

    ConsultationResponse create(CreateConsultationRequest request);

    ConsultationResponse start(Long id);

    ConsultationResponse complete(Long id);

    ConsultationResponse changeStatus(Long id, ChangeConsultationStatusRequest request);
}

File: frontend/src/services/consultationService.js
import api from "./axiosClient";

const consultationService = {
  getAll(params) {
    return api.get("/consultations", { params });
  },

  getById(id) {
    return api.get(`/consultations/${id}`);
  },

  create(data) {
    return api.post("/consultations", data);
  },

  start(id) {
    return api.put(`/consultations/${id}/start`);
  },

  complete(id) {
    return api.put(`/consultations/${id}/complete`);
  },

  changeStatus(id, status) {
    return api.put(`/consultations/${id}/status`, { status });
  },
};

export default consultationService;
